### ISSUE TITLE
Configure window title pattern

### ISSUE THREAD

**Issue by bglw** [zed-industries/zed#14534]
### Check for existing issues

- [X] Completed

### Describe the feature

Currently, Zed's windows titles take the form of `{filename} — {project}`, e.g. `channel.rs — app`.

I would like to configure this to be the inverse: `{project} — {filename}`, e.g. `app — channel.rs`.

I tend to have one window for each project, with multiple projects open at once. Having the project name prefix the filename ensures that in my window switcher, the project name is always visible even when the title is truncated.

As-is, I can wind up with situations like:
- `Cargo.toml — project_a`
- `Cargo.toml — project_b`
- `Cargo.toml — project_c`

Truncated to:
- `Cargo.toml — pr...`
- `Cargo.toml — pr...`
- `Cargo.toml — pr...`

Searching for this, I only turned up #5283 — which should possibly be closed as complete anyway since I _am_ seeing details in the window title.

### If applicable, add mockups / screenshots to help present your vision of the feature

Example of the situation I can currently find myself in when switching windows: (this is the full width of my switcher)

![Screenshot 2024-07-16 at 4 06 55 PM](https://github.com/user-attachments/assets/8c0cfbfc-d953-4db5-a472-3e967008fdc5)

Vision of the feature: my prior VSCode setting:

```
  "window.title": "${rootName}${separator}${activeEditorShort}",
```

**Comment by bglw** [comment]
_(or as a quicker fix, even just an option to omit the filename and show only the project, which saves building any sort of template string parsing)_

**Comment by dmitryrn** [comment]
I stumbled upon a problem when I want to find a Zed window, but I can't just search by Zed, because it's not there. As an example, a VSCodium window has its name in the title, maybe Zed should have it too:
```
wmctrl -l
0x04600003  0 nixos idea.log - log - VSCodium
0x00600003  4 nixos Alacritty
0x0180003d  0 nixos Configure window title pattern · Issue #14534 · zed-industries/zed — Mozilla Firefox
0x03a00003  0 nixos wmctrl -l
0x03c00002  0   N/A settings.json —            <- this is Zed
```

**Comment by nt8r** [comment]
For anyone who might wish to give this a go, the relevant code is [`fn update_window_title`](https://github.com/zed-industries/zed/blob/8603a908c1e97109498c16a7ed60a674be85e352/crates/workspace/src/workspace.rs#L3764) in `crates/workspace/src/workspace.rs`.

**Comment by MaMrEzO** [comment]
Also, it would be nice to have a setting for workspace title, like what Intellij has, for example: 
The project is in `~/Projects/ums`, the workspace title will be `ums`, and I want to set the workspace title to `User Management Service`.

**Comment by an2deg** [comment]
Same here. It would be nice to have a mechanism to set a title for a project. In my case, I have a project with manually added folders (monorepo, too many packages inside, and I only want a subset of those packages).
Zed uses the name of the topmost folder in the list as the title, and this does not reflect the purpose of the project.

**Comment by YMan84** [comment]
> I stumbled upon a problem when I want to find a Zed window, but I can't just search by Zed, because it's not there. As an example, a VSCodium window has its name in the title, maybe Zed should have it too:

Ran into this as well today. All the programs i have open have the program name in the title, Zed is the lone exception. Makes it a real pain to setup keyboard shortcuts to switch to the Zed window.

**Comment by bbb651** [comment]
I'll open a PR hard-coding the `— Zed` clearly there's a need for it, it's a simple`title.push_str(" — "); title.push_str(RELEASE_CHANNEL.display_name())`, should should the collab `↙`/`↗` be before or after it? Imo it can even go after the project name instead of the edited file because it's logically per-project and it's more visible if the title gets cut-off

**Comment by neruthes** [comment]
I use a tiling window manager. Most people stick to the _one-repo, one-directory, one-window_ convention, so the current active filename in the window title is a distraction which I want to completely remove. Basename of the directory is all I need in the window title. BTW I ask the WM to automatically assign windows to fixed virtual desktops depending on their WM_CLASS (`assign [class="dev.zed.Zed"] 2`) so any application name in the window title is also a distraction for me. Hope the maintainers go the vscode approach and allow customizing window title pattern.

<img width="3838" height="339" alt="Image" src="https://github.com/user-attachments/assets/9c383fc2-0841-49aa-b907-c79218222099" />

In the mean time, I asked Gemini to write a small daemon in Python to rename windows and it works on my machine.

https://github.com/neruthes/NDevShellRC/blob/master/bin/myi3companion.py

<img width="3838" height="714" alt="Image" src="https://github.com/user-attachments/assets/e4fa0650-70e8-4c1f-9830-8f5ff30ba070" />

**Comment by jknlsn** [comment]
I put up a PR to implement this here #54379, if anyone has any feedback or wants to try it out! Would be glad to hear any feedback :)

**Comment by prigaux** [comment]
> [@steflsd](https://github.com/steflsd) Good news, they have implemented this. You can open the project settings, under the burger menu, click the `Open Project Settings` item, and you will find `Project Name` input on the first page of settings.

NB: `project_name` has been removed, cf https://github.com/zed-industries/zed/pull/54511

**Comment by MaMrEzO** [comment]
> NB: `project_name` has been removed, cf [#54511](https://github.com/zed-industries/zed/pull/54511)

Yes!!!, and with key binding problems, opening projects in same window(Open folder with Zed command), and many other unexpected behavior, now, I'm deciding to move back on Vim or Intellij...
